src/contracts/permissions/PermissionController.sol

```diff
@@ -0,0 +1,176 @@
+        // If the admin is already set, the set will fail
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+        require(admins.add(admin), AdminAlreadySet());
+
+        emit AdminSet(account, admin);
+    }
+
+    function removeAdmin(address account, address admin) external onlyAdmin(account) {
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+
+        // Remove the admin from the account's admins
+        // If the admin is not set, the remove will fail
+        require(admins.remove(admin), AdminNotSet());
+
+        emit AdminRemoved(account, admin);
+    }
+
+    /// @inheritdoc IPermissionController
+    function setDelegate(
+        address account,
+        address delegate,
+        address target,
+        bytes4 selector
+    ) external onlyAdmin(account) {
+        AccountPermissions storage permissions = _permissions[account];
+
+        bytes32 targetSelector = _encodeTargetSelector(target, selector);
+        require(!permissions.delegatePermissions[delegate].contains(targetSelector), DelegateAlreadySet());
+
+        // Add the delegate to the account's permissions
+        permissions.delegatePermissions[delegate].add(targetSelector);
+        permissions.permissionDelegates[targetSelector].add(delegate);
+
+        emit DelegateSet(account, delegate, target, selector);
+    }
+
+    /// @inheritdoc IPermissionController
+    function removeDelegate(
+        address account,
+        address delegate,
+        address target,
+        bytes4 selector
+    ) external onlyAdmin(account) {
+        AccountPermissions storage permissions = _permissions[account];
+
+        bytes32 targetSelector = _encodeTargetSelector(target, selector);
+        require(permissions.delegatePermissions[delegate].contains(targetSelector), DelegateNotSet());
+
+        // Remove the delegate from the account's permissions
+        permissions.delegatePermissions[delegate].remove(targetSelector);
+        permissions.permissionDelegates[targetSelector].remove(delegate);
+
+        emit DelegateRemoved(account, delegate, target, selector);
+    }
+
+    /**
+     *
+     *                         INTERNAL FUNCTIONS
+     *
+     */
+
+    /// @dev Encodes the target and selector into a single bytes32 value
+    function _encodeTargetSelector(address target, bytes4 selector) internal pure returns (bytes32) {
+        return bytes32(abi.encodePacked(target, uint96(bytes12((selector)))));
+    }
+
+    /// @dev Decodes the target and selector from a single bytes32 value
+    function _decodeTargetSelector(
+        bytes32 targetSelector
+    ) internal view returns (address, bytes4) {
+        address target = address(uint160(uint256(targetSelector) >> 96));
+        // The selector is in the lower 32 bits of the targetSelector
+        bytes4 selector = bytes4(uint32(uint256(targetSelector) >> 64));
+
+        return (target, selector);
+    }
```

Comment:
If we are using shifts to decode, we should consider using shifts to encode. Makes it easier to reason that they're inverse operations.

Reply:
Reads cleaner: https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/cd629bed6e808752567b701a1fc607d0f9281d14